### PR TITLE
[CSL-2027] Use servant-generic

### PR DESCRIPTION
**Pull request by neongreen**

### CODE REVIEW

**Comment by neongreen** [comment]
Don't merge it yet, I want to convert the wallet as well.

File: stack.yaml [hunk 1]
```diff
@@ -111,6 +111,11 @@ packages:
     commit: 2d261bb971bada1893753b503452d9e6e217bc4a
   extra-dep: true
 
+- location:
+    git: https://github.com/neongreen/servant-generic.git
```
Comment (gromakovsky) [inline stack.yaml]: Maybe move it to `serokell/` to make our lives potentially easier if you are hit by a bus?

**Comment by adinapoli-iohk** [comment]
> Don't merge it yet, I want to convert the wallet as well.

@neongreen Can I kindly ask you to do that as a separate PR? 😉  That might make a bit easier for us poor devs working on the moving target `feature/new-wallet` is handle the breaking changes, if any :)

**Comment by neongreen** [comment]
> Maybe move it to serokell/ to make our lives potentially easier if you are hit by a bus?

fine, since two people asked for it I'll do it

**Comment by neongreen** [comment]
> Can I kindly ask you to do that as a separate PR?

okay, then in an hour I'll fix default.nix and remove the WIP flag from this PR

File: stack.yaml (hunk 1 above)
Comment (neongreen) [reply stack.yaml]: done

**Comment by ch1bo** [comment]
Hey,

sorry to chime in on your "internal" development here. But I was wondering what the benefit of record syntax data type APIs are? Also, how is route precedence treated?

Again, I'm just curious what your rationale is and maybe I should use servant-generic for my projects

**Comment by neongreen** [comment]
@ch1bo the rationale is:

* You can use named fields for handlers & handler types and not be afraid of messing up the order
* When the handler doesn't match the needed type, you get a much smaller error message (`field blah ... doesn't match blah`)

We don't have overlapping routes AFAIK, but even if we did, the translation from records is very straightforward and `servant-generic` will handle them in the same way as `servant` does.

```haskell
data Foo route = Foo                type Foo 
  { a :: route :- A                     = A    
  , b :: route :- B                  :<|> B
  , c :: route :- C                  :<|> C
  }
```

**Pull request merged**